src/adcs_io.c

```diff
@@ -38,8 +39,25 @@ ADCS_returnState send_uart_telecommand(uint8_t* command, uint32_t length) {
   return TC_err_flag;
 }
 
-ADCS_returnState send_i2c_telecommand(uint8_t* command, uint32_t length) {
-  return ADCS_OK;
+ADCS_returnState send_i2c_telecommand(uint8_t *command, uint32_t length)
+{
+  // Send telecommand
+  i2c_send(command, length);
```

Comment:
Don't we need some sort of processing on the I2C frame that we're sending? I see some address bytes and stop bits. For example for UART, the frame is finalized and ready to be sent over the connection.

Reply:
So, we need to set a start condition and send a byte to initiate an i2c write operation and also set a stop condition. The i2c drivers that HalCoGen generates have functions that accomplish this (such as i2cSetStart, i2cSetMode, i2cIsTxReady etc). 

I was thinking that maybe the i2c_send function in uart_i2c.h would be implemented to call these HalCoGen functions, and handle that for us? Also, is that abstraction necessary? In the future, after getting rid of the mocks, could we directly call and use the HalCoGen driver functions in these adcs_io.c functions?

Reply:
Well, it's your decision but either should be fine. If you think it'd be better to differentiate the abstractions, you can add an `adcs_i2c` file and do that processing in there, but you can definitely develop that even if we mock it in the tests right now. It'd make it easier for the people integrating it with hardware. Although, probably they'd require to change some stuff to make it work.

Reply:
Hmm. I think I will just implement those functions that are being mocked right now. I don't think this would interfere with ceedling, and would also make the code more complete.